pkg/diagnose/runner.go

```diff
@@ -333,8 +303,8 @@ func requestDiagnosesFromAgentProcess(diagCfg diagnosis.Config) ([]diagnosis.Dia
 	}
 
 	// Deserialize results
-	var diagnoses []diagnosis.Diagnoses
-	err = json.NewDecoder(bytes.NewReader(response)).Decode(&diagnoses)
+	var diagnoses *diagnosis.DiagnoseResult
+	err = json.Unmarshal(response, &diagnoses)
```

Comment:
Are you sure about this line? 

I see that `diagnoses` is already a pointer, why are we unmarshalling onto &diagnoses and not diagnosis?

Reply:
good catch just fixed it